Lib/ensurepip/__init__.py

```diff
@@ -27,8 +27,8 @@ def _run_pip(args, additional_paths=None):
         sys.path = additional_paths + sys.path
 
     # Install the bundled software
-    import pip._internal
-    return pip._internal.main(args)
+    import pip._internal.main
+    return pip._internal.main.main(args)
```

Comment:
Are we OK to change this to use runpy? That way we'd basically be doing python -m pip, which is directly supported.

See https://github.com/pradyunsg/pip-cli/blob/master/src/_pip_cli.py#L75 for how we can do that. You can skip setting the env there, since that's for passing information that's not relevant here.

Reply:
I wouldn't say that it's directly supported (as in, part of the public API) since running in the same process space is still not considered supported, but it would avoid any issues with changes to the location of the entrypoint function.

It may also be helpful to consider backporting that change to bugfix Python versions, which can help people that build and patch in new versions of pip (as mentioned in 09fd200).

Reply:
Why is this not simply run in a sub-process, then?

Reply:
We don't use runpy (or a subprocess) here because of the path manipulation just above to import pip from the bundled wheel archives.

It's useful to think of ensurepip as being part of pip, as we're free to make whatever changes are needed to sync up when updating to new versions of pip.

Reply:
Let's discuss in https://bugs.python.org/issue38662 or #17029

Reply:
https://bugs.python.org/issue38662 is now fixed in master. You can rebased your PR on master.

Reply:
Done.